deploy/stacks/waf_rules.py

```diff
@@ -0,0 +1,179 @@
+                    vendor_name='AWS', name='AWSManagedRulesAdminProtectionRuleSet'
+                )
+            ),
+            visibility_config=wafv2.CfnWebACL.VisibilityConfigProperty(
+                sampled_requests_enabled=True,
+                cloud_watch_metrics_enabled=True,
+                metric_name='AWS-AWSManagedRulesAdminProtectionRuleSet',
+            ),
+            priority=priority,
+            override_action=wafv2.CfnWebACL.OverrideActionProperty(none={}),
+        )
+    )
+    priority += 1
+    waf_rules.append(
+        wafv2.CfnWebACL.RuleProperty(
+            name='AWS-AWSManagedRulesAmazonIpReputationList',
+            statement=wafv2.CfnWebACL.StatementProperty(
+                managed_rule_group_statement=wafv2.CfnWebACL.ManagedRuleGroupStatementProperty(
+                    vendor_name='AWS', name='AWSManagedRulesAmazonIpReputationList'
+                )
+            ),
+            visibility_config=wafv2.CfnWebACL.VisibilityConfigProperty(
+                sampled_requests_enabled=True,
+                cloud_watch_metrics_enabled=True,
+                metric_name='AWS-AWSManagedRulesAmazonIpReputationList',
+            ),
+            priority=priority,
+            override_action=wafv2.CfnWebACL.OverrideActionProperty(none={}),
+        )
+    )
+    priority += 1
+    waf_rules.append(
+        wafv2.CfnWebACL.RuleProperty(
+            name='AWS-AWSManagedRulesCommonRuleSet',
+            statement=wafv2.CfnWebACL.StatementProperty(
+                managed_rule_group_statement=wafv2.CfnWebACL.ManagedRuleGroupStatementProperty(
+                    vendor_name='AWS', name='AWSManagedRulesCommonRuleSet'
+                )
+            ),
+            visibility_config=wafv2.CfnWebACL.VisibilityConfigProperty(
+                sampled_requests_enabled=True,
+                cloud_watch_metrics_enabled=True,
+                metric_name='AWS-AWSManagedRulesCommonRuleSet',
+            ),
+            priority=priority,
+            override_action=wafv2.CfnWebACL.OverrideActionProperty(none={}),
+        )
+    )
+    priority += 1
+    waf_rules.append(
+        wafv2.CfnWebACL.RuleProperty(
+            name='AWS-AWSManagedRulesKnownBadInputsRuleSet',
+            statement=wafv2.CfnWebACL.StatementProperty(
+                managed_rule_group_statement=wafv2.CfnWebACL.ManagedRuleGroupStatementProperty(
+                    vendor_name='AWS', name='AWSManagedRulesKnownBadInputsRuleSet'
+                )
+            ),
+            visibility_config=wafv2.CfnWebACL.VisibilityConfigProperty(
+                sampled_requests_enabled=True,
+                cloud_watch_metrics_enabled=True,
+                metric_name='AWS-AWSManagedRulesKnownBadInputsRuleSet',
+            ),
+            priority=priority,
+            override_action=wafv2.CfnWebACL.OverrideActionProperty(none={}),
+        )
+    )
+    priority += 1
+    waf_rules.append(
```

Comment:
I have been searching for recommended WAF rules for Cognito in particular. Because it is a recent launch the resources are a bit limited, but in general geo-fencing, IP-allow-list, rate limits and the typical AWSManagedRulesCommonRuleSet appear in demos and blogs.
From a cost perspective, WAF currently adds $23/month to the data.all bill. But if you see the docs: https://aws.amazon.com/waf/pricing/ going over 1500 WCUs affects the costs slightly `The total WCUs for a web ACL can't exceed 5000. Using over 1500 WCUs affects your costs.` we are currently at 1427.

In short:
- there are no additional WAF rules to add at the moment
- we might remove the SQL or the linux rules, but it is not a big deal because the costs are very low. wdyt?

Reply:
Agreed - I could not find any prescriptive guidance on best practice WAF rules for Cognito. I think we are okay to keep rules the same for now and if needed break it out so each WebACL has its own unique rule set at a later point